### PR TITLE
Add early stopping in BiLSTM and CNN

### LINKED ISSUE #139: CNN and BiLSTM do not support early stopping
Both CNN and BiLSTM currently run training for as many epochs as the user has instructed with no ability to stop when the loss stops decreasing.

Tensorflow provides a callback for that which can easily be added and also allows to restore best weights.

### PR DESCRIPTION
**Pull request by nsorros**
Description
---
Fixes #139 

Adds param early_stopping with a default patience of 5 epochs and default ability to restore best weights. In the future these can also be params.

Checklist
---

- [x] Added link to Github issue or Trello card
- [x] Added tests

### CODE REVIEW

File: wellcomeml/ml/bilstm.py [hunk 1]
```diff
@@ -133,6 +135,10 @@ def fit(self, X, Y, embedding_matrix=None, *_):
             CALLBACK_DICT[c] if c in CALLBACK_DICT else c
             for c in self.callbacks
         ]
+        if self.early_stopping:
+            early_stopping = tf.keras.callbacks.EarlyStopping(
```
Comment (aCampello) [inline wellcomeml/ml/bilstm.py:139]: That is such a useful callback - I did not know that. Does it just monitor the training loss, or anyway you can make it monitor validation too?

**Review by aCampello — APPROVED** [review]

File: wellcomeml/ml/bilstm.py (hunk 1 above)
Comment (nsorros) [reply wellcomeml/ml/bilstm.py:139]: It monitors validation loss by default but you can change to any of the available metrics https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/EarlyStopping
